File: app/src/main/java/io/mgba/DI/Components/ModelComponent.java
package io.mgba.DI.Components;

import javax.inject.Singleton;

import dagger.Component;
import io.mgba.DI.Modules.ModelModule;
import io.mgba.Model.Library;
import io.mgba.Presenter.IntroPresenter;
import io.mgba.Presenter.MainPresenter;

@Singleton
@Component(modules = ModelModule.class)
public interface ModelComponent {
    void inject(IntroPresenter target);
    void inject(MainPresenter target);
    void inject(Library library);
}

File: app/src/main/java/io/mgba/di/components/ModelComponent.kt
package io.mgba.di.components

import javax.inject.Singleton

import dagger.Component
import io.mgba.di.modules.ModelModule
import io.mgba.model.Library
import io.mgba.presenter.IntroPresenter
import io.mgba.presenter.MainPresenter

@Singleton
@Component(modules = [ModelModule::class])
interface ModelComponent {
    fun inject(target: IntroPresenter)
    fun inject(target: MainPresenter)
    fun inject(library: Library)
}
